### DISCUSSION TITLE
Preview deployment is coming on google search

### DISCUSSION THREAD

**Discussion by rishipurwar1** [vercel/community#1848] [Help] [answered]
### Summary

Hello there,

Yesterday, I changed one of the vercel URLs from production to preview, but it's still coming on google search instead of my main custom domain. Could you please help me fix that issue?

Please let me know if you need more info!

Thanks,
Rishi

### Example

https://coding-space.vercel.app

**Comment by amyegan** [comment]
Hi Rishi. Can you share a screenshot so I can see what you mean about a preview deployment coming up on Google search instead of your custom domain?

**Reply by rishipurwar1** [reply]
Check out this screenshot:

<img width="553" alt="Screenshot 2023-03-22 at 10 07 29 PM" src="https://user-images.githubusercontent.com/57035408/226975130-3b7b2e6e-26e2-4519-8b9e-7fcd7e82d3df.png">


According to [this doc](https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines): "Vercel Preview Deployments are not indexed by search engines by default because the X-Robots-Tag HTTP header is set to noindex" but still this preview domain is coming up on google search.

<img width="626" alt="Screenshot 2023-03-22 at 10 09 34 PM" src="https://user-images.githubusercontent.com/57035408/226976326-d8f5a441-6775-4c39-bbcd-b5067804da1d.png">

**Reply by rishipurwar1** [reply]
Because of this preview domain, my custom domain(production) is not able to get indexed by search engines properly.

**Reply by amyegan** [reply]
Thanks for clarifying that, Rishi. Can you tell me how long it has been since making this domain setting change? And have you redeployed your latest `development` branch preview and `main` branch production deployments since making the change? I'm trying to rule out things like cache and DNS propagation delays

**Reply by rishipurwar1** [reply]
I made changes to the domain setting yesterday and subsequently redeployed both the preview and main after implementing those modifications.

**Comment by amyegan** [comment, marked as answer]
The `coding-space.vercel.app` is being treated as a custom domain, since you have the ability to configure it like your other custom domains. This is slightly different from a normal preview domain.

> If you are using a [Custom Domain](https://vercel.com/docs/custom-domains) that is assigned to a non-[Production Branch](https://vercel.com/docs/git#production-branch), however, the header X-Robots-Tag: noindex will not be set.   — [X-Robots-Tag Header](https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines#x-robots-tag-header)

That means you need to explicitly configure your preview deployments with custom domains to include the `X-Robots-Tag: noindex` header. For example, if you use Next.js then you could set the header for all preview deployments in your `next.config.js` file, like: 
```js
module.exports = {
  async headers() {
    const headers = [];
    if (process.env.NEXT_PUBLIC_VERCEL_ENV === 'preview') {
      headers.push({
        headers: [
          {
            key: 'X-Robots-Tag',
            value: 'noindex',
          },
        ],
        source: '/:path*',
      });
    }
    return headers;
  },
 };
```

**Reply by rishipurwar1** [reply]
I just added the headers in the next.config.js file like this:
```
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ["res.cloudinary.com", "avatars.githubusercontent.com"],
  },
  async headers() {
    const headers = []
    if (process.env.NEXT_PUBLIC_VERCEL_ENV === "preview") {
      headers.push({
        headers: [
          {
            key: "X-Robots-Tag",
            value: "noindex",
          },
        ],
        source: "/:path*",
      })
    }
    return headers
  },
}

module.exports = nextConfig
```

Does it take some time to be configured properly, I also don't have`NEXT_PUBLIC_VERCEL_ENV` environment variable in my .env file, does vercel automatically takes care of that?

**Reply by amyegan** [reply]
`NEXT_PUBLIC_VERCEL_ENV` is a [Framework Environment Variable](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#framework-environment-variables) provided by Vercel for Next.js projects. After deploying with the new config, you can check for the header using `curl -I https://coding-space.vercel.app`. I do see the header coming through now, so it looks like your config change is working. 🎉 

You may also need to request that search engines, like Google, [recrawl your site](https://developers.google.com/search/docs/crawling-indexing/ask-google-to-recrawl) and [remove outdated content from search results](https://support.google.com/websearch/answer/11080680?hl=en)

**Reply by rishipurwar1** [reply]
Okay, thanks for the help!
